tools/CheckMetadata.py

```diff
@@ -0,0 +1,179 @@
+                                      self.metadata['Doi']).group(1)
+        self.assertEqual(zenodo_id, int(zenodo_id_from_doi),
+                         "Zenodo ID should match DOI")
+
+    def test_description(self):
+        self.assertIn('Description', self.metadata)
+        self.assertTrue(isinstance(self.metadata['Description'], str))
+
+    def test_keywords(self):
+        self.assertIn('Keywords', self.metadata)
+        keywords = self.metadata['Keywords']
+        self.assertTrue(isinstance(keywords, list))
+        self.assertTrue([isinstance(keyword, str) for keyword in keywords])
+
+    def test_affiliations(self):
+        self.assertIn('Affiliations', self.metadata)
+        affiliations = self.metadata['Affiliations']
+        self.assertTrue(isinstance(affiliations, list))
+        self.assertTrue(
+            [isinstance(affiliation, str) for affiliation in affiliations])
+        # Check for duplicates
+        simplified_affiliations = [
+            simplify_name(affiliation) for affiliation in affiliations
+        ]
+        for affiliation in affiliations:
+            self.assertTrue(
+                simplified_affiliations.count(simplify_name(affiliation)) == 1,
+                f"Duplicate affiliation: {affiliation}")
+
+    def test_authors(self):
+        self.assertIn('Authors', self.metadata)
+        authors = self.metadata['Authors']
+        self.assertTrue(isinstance(authors['Description'], str))
+        authors_core = authors['Core']
+        self.assertTrue(isinstance(authors_core['Description'], str))
+        authors_core_list = authors_core['List']
+        self.assertTrue(isinstance(authors_core_list, list))
+        authors_devs = authors['Developers']
+        self.assertTrue(isinstance(authors_devs['Description'], str))
+        authors_devs_list = authors_devs['List']
+        self.assertTrue(isinstance(authors_devs_list, list))
+        authors_contribs = authors['Contributors']
+        self.assertTrue(isinstance(authors_contribs['Description'], str))
+        authors_contribs_list = authors_contribs['List']
+        self.assertTrue(isinstance(authors_contribs_list, list))
+        all_authors = (authors_core_list + authors_devs_list +
+                       authors_contribs_list)
+
+        # Check alphabetical ordering
+        self.assertEqual(
+            sorted(authors_devs_list,
+                   key=lambda a: a['Name']), authors_devs_list,
+            "'Developers' author list should be ordered alphabetically")
+        self.assertEqual(
+            sorted(authors_contribs_list,
+                   key=lambda a: a['Name']), authors_contribs_list,
+            "'Contributors' author list should be ordered alphabetically")
+
+        # Check all authors
+        def check_name(name):
+            split_name = name.split(', ')
+            self.assertTrue(
+                len(split_name) == 2,
+                f"Name '{name}' should be formatted 'Last name, First name'")
```

Comment:
Should we handle "Jr.", etc.?  (BibTeX uses "Last, Jr., First".)

Reply:
I think it would be great if we could offer it, but Zenodo only supports `Last Name, First Name` (or at least that's what their documentation says...). However, given that the Zenodo documentation isn't always up-to-date I think it would be good to test to see what happens if we just add a contributor "Last, Jr., First" and see what Zenodo does in the sandbox. @nilsleiffischer would you mind testing that?

Reply:
Seems fine, at least nothing obviously breaks. In fact it seems like Zenodo doesn't process the name at all, it just appears in the `Last[, Jr.], First` format in the author list of the publication and in the citation suggestions, including the BibTeX entry. I adjusted the test to allow for the optional name component.

Reply:
Okay, awesome! Thanks so much :)